src/lib.rs

```diff
@@ -139,6 +139,16 @@ pub struct Metadata {
     __do_not_match_exhaustively: (),
 }
 
+impl Metadata {
+    /// Get the root package of this metadata instance.
+    pub fn root_package(&self) -> Option<&Package> {
+        self.resolve
+            .as_ref()
+            .and_then(|res| res.root.as_ref())
+            .and_then(|root| self.packages.iter().find(|pkg| &pkg.id == root))
```

Comment:
This could also use the `?` operator instead of `and_then`:

```suggestion
        let res = self.resolve.as_ref()?;
        let root = res.root.as_ref()?;
        self.packages.iter().find(|pkg| &pkg.id == root)
```

Reply:
Np. If you prefer that style, I'll do that.